internal/evaluator/evaluator.go

```diff
@@ -0,0 +1,78 @@
+package evaluator
+
+import (
+	"context"
+	"github.com/cirruslabs/cirrus-ci-agent/api"
+	"github.com/cirruslabs/cirrus-cli/pkg/larker"
+	"github.com/cirruslabs/cirrus-cli/pkg/parser"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"net"
+	"strings"
+)
+
+func Serve(ctx context.Context, lis net.Listener) error {
+	server := grpc.NewServer()
+
+	api.RegisterCirrusConfigurationEvaluatorServiceService(server, &api.CirrusConfigurationEvaluatorServiceService{
+		EvaluateConfig: evaluateConfig,
+	})
+
+	errChan := make(chan error)
+
+	go func() {
+		errChan <- server.Serve(lis)
+	}()
+
+	select {
+	case <-ctx.Done():
+		server.GracefulStop()
+
+		if err := <-errChan; err != nil {
+			return err
+		}
+
+		return ctx.Err()
+	case err := <-errChan:
+		return err
+	}
+}
+
+func evaluateConfig(ctx context.Context, request *api.EvaluateConfigRequest) (*api.EvaluateConfigResponse, error) {
+	var yamlConfigs []string
+
+	// Register YAML configuration (if any)
+	if request.YamlConfig != "" {
+		yamlConfigs = append(yamlConfigs, request.YamlConfig)
+	}
+
+	// Run Starlark script and register generated YAML configuration (if any)
+	if request.StarlarkConfig != "" {
+		lrk := larker.New(larker.WithEnvironment(request.Environment))
+
+		generatedYamlConfig, err := lrk.Main(ctx, request.StarlarkConfig)
+		if err != nil {
+			return nil, status.Error(codes.InvalidArgument, err.Error())
+		}
+
+		yamlConfigs = append(yamlConfigs, generatedYamlConfig)
+	}
+
+	// Parse combined YAML
+	p := parser.New(
+		parser.WithEnvironment(request.Environment),
+		parser.WithFilesContents(request.FilesContents),
```

Comment:
Let's also add GitHub FS in case environment contains `CIRRUS_REPO_CLONE_TOKEN`, `CIRRUS_REPO_NAME`, `CIRRUS_REPO_OWNER` and `CIRRUS_CHANGE_IN_REPO`. Seems this will require the parser to check explicit file contents first and if a file is not presented then check in fs.

Reply:
Implemented in https://github.com/cirruslabs/cirrus-cli/pull/114/commits/258567df788f30780ecc1894c8c3cc8f96f400c7.

I'm not sure why we need to touch files contents or parser at all, though, since GitHub FS is only used by Starlark.